erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -871,16 +871,25 @@ frappe.ui.form.on('Payment Entry', {
 			function(d) { return flt(d.amount) }));
 
 		if(frm.doc.party) {
-			if(frm.doc.payment_type == "Receive"
+			if(frm.doc.payment_type === "Receive"
 				&& frm.doc.base_total_allocated_amount < frm.doc.base_received_amount + total_deductions
 				&& frm.doc.total_allocated_amount < frm.doc.paid_amount + (total_deductions / frm.doc.source_exchange_rate)) {
-					unallocated_amount = (frm.doc.base_received_amount + total_deductions + frm.doc.base_total_taxes_and_charges
-						- frm.doc.base_total_allocated_amount) / frm.doc.source_exchange_rate;
-			} else if (frm.doc.payment_type == "Pay"
+					unallocated_amount = (
```

Comment:
Need to look into this again. Just found out Allocated credit notes is not working as expected